### PR TITLE
update R API documentation for `mlflow_register_external_observer()`

### PR DESCRIPTION
**Pull request by yitao-li**
Signed-off-by: Yitao Li <yitao@rstudio.com>

## What changes are proposed in this pull request?

Fixing the missing update to R API documentation from https://github.com/mlflow/mlflow/pull/3501

## How is this patch tested?

- Checking changes in the docs/source/R-api.rst file make sense
- Waiting for all existing GitHub CI workflows for `mlflow` to complete successfully

### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [x] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?
Components 
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [x] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: Local serving, model deployment tools, spark UDFs
- [ ] `area/server-infra`: MLflow server, JavaScript dev server
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface 
- [ ] `area/uiux`: Front-end, user experience, JavaScript, plotting
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language 
- [x] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

**Review by javierluraschi — APPROVED** [review]

**Comment by javierluraschi** [comment]
R test seems to be reliable breaking, maybe consider rebasing from master?

```
2020/10/20 04:32:50 INFO mlflow.projects.utils: === Created directory /tmp/tmpsh2vzwc5 for downloading remote URIs passed to arguments of type 'path' ===
2020/10/20 04:32:50 INFO mlflow.projects.backend.local: === Running command 'source /usr/share/miniconda/bin/../etc/profile.d/conda.sh && conda activate mlflow-da39a3ee5e6b4b0d3255bfef95601890afd80709 1>&2 && Rscript -e "mlflow::mlflow_source('params_example.R')" --args --my_int 10 --my_num 20.0 --my_str XYZ' in run with ID '5dcb1b39ef1b415aa506f665edd8fcde' === 
Error: object of type 'closure' is not subsettable
```

**Comment by harupy** [comment]
Rebasing should solve the error. #3544 fixed it.

**Comment by yitao-li** [comment]
R tests are passing now after rebase. Thanks @harupy !!

File: docs/source/R-api.rst [hunk 1]
```diff
@@ -1058,7 +1104,7 @@ restored.
 
    mlflow_restore_experiment(experiment_id, client = NULL)
 
-.. _arguments-26:
+.. _arguments-27:
```
Comment (harupy) [inline docs/source/R-api.rst:1107]: I've been wondering why we always have to increment argument numbers like this everytime we add a new function.

Comment (harupy) [reply docs/source/R-api.rst:1107]: and what would happen if we removed these?

Comment (yitao-li) [reply docs/source/R-api.rst:1107]: @harupy The changes are generated by a script (not entirely sure if that answers the question, but anyhow, that's about as much as I know 🙂)

Comment (harupy) [reply docs/source/R-api.rst:1107]: Which script? `document.R`?

Comment (yitao-li) [reply docs/source/R-api.rst:1107]: Yes indeed. It's whatever the docs/build-rdoc.sh script does, and looks like most of it is running Rscript document.R

**Pull request merged**